app/models/genre_term.rb

```diff
@@ -0,0 +1,93 @@
+# Copyright 2017, University of Alberta Libraries.
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+class GenreTerm
+  require 'avalon/levenshtein'
+  extend Avalon::Levenshtein
+
+  class LookupError < Exception; end
+  class DuplicateKeyError < Exception; end
+
+  STORE = File.join(Rails.root, 'config/pbcore_genre.csv')
+
+  class << self
```

Comment:
Not a huge fan either, but following what upstream is doing. In particular, the structure of this file mirrors very closely this one:

https://github.com/ualbertalib/avalon/blob/master/app/models/language_term.rb